app/Controllers/CtrlSolution.php

```diff
@@ -20,114 +39,176 @@ public function viewSolutions()
             ];
 
-            return redirect()->to('admin/soluciones')->with('response', $response);
+            return redirect()->to(self::MEASUREMENT_SOLUTIONS_BASE_ROUTE)->with('response', $response);
         } catch (InvalidInputException $th) {
-            return redirect()->to('admin/soluciones/crear')->withInput()->with('errors', $th->getErrors());
+            session()->setFlashdata('clientData', $msData);
+
+            return redirect()->to(self::MEASUREMENT_SOLUTIONS_CREATE_ROUTE)->withInput()->with('errors', $th->getErrors());
         } catch (Throwable $th) {
+            session()->setFlashdata('clientData', $msData);
             $response = [
                 'title'   => 'Oops! Ha ocurrido un error.',
                 'message' => 'Ha ocurrido un error al registrar los datos, por favor intente nuevamente.',
                 'type'    => 'error',
             ];
 
-            return redirect()->to('/admin/soluciones/crear')->with('response', $response)->withInput();
+            return redirect()->to(self::MEASUREMENT_SOLUTIONS_CREATE_ROUTE)->withInput()->with('response', $response);
         }
     }
 
     public function updateSolution(string $msId)
     {
+        $msUpdatedData = $this->request->getPost();
+
         try {
-            $msData = $this->request->getPost();
+            $fileValidationRules = $this->fileConfig->getCollectionFileValidationRules();
 
             $msDataValidator = new SolutionValidation();
+            $msDataValidator->addRules($fileValidationRules['rules'], $fileValidationRules['messages']);
 
-            if (! $msDataValidator->validateInputs($msData)) {
+            if (! $msDataValidator->validateInputs($msUpdatedData)) {
                 throw new InvalidInputException($msDataValidator->getErrors());
             }
 
-            $msModel = new MeasurementSolutionModel();
-            $msModel->updateMeasurementSolution($msId, $msData);
+            $this->insertMsFilesData($msUpdatedData);
+
+            (new MeasurementSolutionModel())->updateMeasurementSolution($msId, $msUpdatedData);
+
+            $this->storeMsFiles($msUpdatedData);
+
+            $this->deletePreviousMsFiles($msUpdatedData);
 
             $response = [
                 'title'   => 'Actualización exitosa',
                 'message' => 'Los datos de la solución de medición han sido actualizados correctamente',
                 'type'    => 'success',
             ];
 
-            return redirect()->to('/admin/soluciones')->with('response', $response);
+            return redirect()->to(self::MEASUREMENT_SOLUTIONS_BASE_ROUTE)->with('response', $response);
         } catch (InvalidInputException $th) {
-            return redirect()->back()->withInput()->with('errors', $th->getErrors());
+            session()->setFlashdata('clientData', $msUpdatedData);
+
+            return redirect()->to(self::MEASUREMENT_SOLUTIONS_EDIT_ROUTE . $msId)->withInput()->with('errors', $th->getErrors());
```

Comment:
Here, in order you use only once the `MEASUREMENT_SOLUTIONS_BASE_ROUTE` variable, then you can do here
```suggestion
            return redirect()->to(url_to(self::class . '::viewSolutionEdit', $msId))->withInput()->with('errors', $th->getErrors());
```

Reply:
why not using the static constant generated above?

Reply:
When we use constants like the way you do before, we can´t use functions in constants expresions
Like: 
```php
private const MS_BASE = url_to(...)
```  
That is incorrect because is not a constant argument, but a function. That´s the reason we need to set an static variable and set it into the `__construct()` function